app/move/controllers/create/assessment.js

```diff
@@ -31,19 +32,26 @@ class AssessmentController extends CreateBaseController {
     this.use(this.setPreviousAssessment)
   }
 
-  setPreviousAssessment(req, res, next) {
+  async setPreviousAssessment(req, res, next) {
     const {
       assessmentCategory,
       customAssessmentGroupings = [],
       showPreviousAssessment,
     } = req.form.options
 
-    if (showPreviousAssessment) {
+    let profile = req.getProfile()
+
+    if (!profile.id) {
       const person = req.getPerson()
+      profile = await profileService.create(person.id, {})
```

Comment:
I think we should create it at the same point at which we create a person. That way it's the safest point in time. As soon as we know we have a person, we create the profile for this move.

That would save you having to check for the id and prevent the issue I'm sure you saw where it would create a new one for each assessment step...

Reply:
Creating it at the point of having a person also ensures that if we use anything else on a profile in the future it will exist rather than being tied to only these steps.

For example, there are also details on the move that only make sense when there is a _person_ so in future these could be moved to the profile to make it cleaner to remove a person and their specific information for that move from the move. If we create the profile sooner, it would ensure that wouldn't catch us out again in future.

Reply:
But, that's exactly why I put it here. We have journeys that create a person and journeys that don't. That's why I think this is the safest point to create the profile.

Would you prefer to create the profile?

a) on the move details page
b) i) when the person is selected from the results / ii) when the person is created on the person-details page